pyrsia_node/src/artifact_manager.rs

```diff
@@ -223,7 +223,7 @@ impl<'a> Write for WriteHashDecorator<'a> {
 
     fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
         let result = self.writer.write(buf);
-        if let Ok(_) = result {
+        if result.is_ok() {
```

Comment:
This could be simplified even more using the `if let` construct:
```suggestion
        if let Ok(_) = self.writer.write(buf) {
```

Reply:
@tiainen Good suggestion.